common/azure-common/src/main/java/com/azure/common/annotations/UnexpectedResponseExceptionType.java

```diff
@@ -1,37 +1,51 @@
-/**
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- */
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
 
 package com.azure.common.annotations;
 
 import com.azure.common.exception.ServiceRequestException;
 
+import java.lang.annotation.Repeatable;
 import java.lang.annotation.Retention;
 import java.lang.annotation.Target;
 
 import static java.lang.annotation.ElementType.METHOD;
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
 /**
- * The error type that will be thrown or returned when an unexpected status code is returned from an REST API.
+ * The exception type that is thrown or returned when one of the status codes is returned from a REST API. Multiple
+ * annotations can be used, they are processed in a top-down order. When no codes are listed that exception is always
+ * thrown or returned if it is reached during evaluation, this should be treated as a default case. If no default case
+ * is annotated the fall through exception is ServiceRequestException.
```

Comment:
Don't fully agree on this approach. I would look up specifically for a stated status code. If that is not found, then there should only be one annotation specified without a code, so you use that (if there are more than one without a code specified, you should throw an exception)

Reply:
Updated how the implementation works to throw an exception when multiple default exceptions are seen or when a HTTP status code is shared across annotations, update the JavaDoc to reflect that.